### DISCUSSION TITLE
Images module backward compatibility, upgrade help

### DISCUSSION THREAD

**Discussion by pensivedog** [gethyas/hyas#439] [Q&A]
I am trying to upgrade a Hyas site to use the latest Images module. After following the upgrade guide, builds fail with errors related to image module partial and shortcode calls. I have many articles using the old shortcode syntax, and many templates using the old partial syntax.

Is the only way to upgrade to manually change each article that uses the shortcode and layout template that calls the partial?

**Comment by pensivedog** [comment]
Turns out I was able to get builds working again by simply doing a find and replace in my template files:

`partial "picture"` to  `partial "picture.html"`

and

`"ctx" .` to `"page" .`

**Reply by pensivedog** [reply]
Actually, the above only worked in dev. I'm still getting an error when running `npm run build`:

```
execute of template failed: template: partials/picture.html:202:10: executing "partials/picture.html" at <site>: can't evaluate field landscape in type bool
```

**Comment by h-enk** [comment]
Did you follow the [Manual Setup](https://images.gethyas.com/docs/start-here/manual-setup/) guide?

Here's a minimum and maximum example of the [picture partial](https://images.gethyas.com/docs/guides/partials/picture/).

Also, check out the full [reference](https://images.gethyas.com/docs/reference/partials/picture/)

**Reply by pensivedog** [reply]
Yes, I followed the Manual Setup guide and have tried this with just the minimal example of the picture partial. In my `articles/single.html` template I have the following:

```
{{ $image := .Page.Resources.GetMatch .Params.featured_image }}
{{- partial "picture.html" (dict "page" . "src" $image) }}
```
This gives the following error when running `npm run build`:

```
ERROR render of "page" failed: "C:\mysite\layouts\articles\single.html:21:16": execute of template failed: template: articles/single.html:21:16: executing "main" at <partial "picture.html" (dict "page" . "src" $image)>: error calling partial: "C:\mysite\node_modules\@hyas\images\layouts\partials\picture.html:202:10": execute of template failed: template: partials/picture.html:202:10: executing "partials/picture.html" at <site>: can't evaluate field landscape in type bool
```

**Reply by h-enk** [reply]
No need to get the image yourself anymore (the logic is in the module). Store your image as a page resource, global resource, or remote resource, and just use [the correct path](https://images.gethyas.com/docs/guides/resources/)

**Reply by h-enk** [reply]
This should work though (when you place your image in `assets/images`:

```html
{{ $image := resources.Get "images/vincent-van-zalinge-IicWDdQUfsQ-unsplash.jpg" }}
{{- partial "picture" (dict "page" . "src" $image) }}
```

**Reply by pensivedog** [reply]
Getting closer... I did find-replace in my layout templates and changed all `{{ $image := .Page.Resources.GetMatch .Params.featured_image }}` to `{{ $image := resources.Get .Params.featured_image }}`.

Now, I no longer get errors for the partial, but I am getting errors for the figure shortcode:

```
Error: error building site: "C:\mysite\content\articles\article-title\index.md:242:1": failed to render shortcode "figure": failed to process shortcode: "C:\mysite\node_modules\@hyas\images\layouts\shortcodes\figure.html:125:4": execute of template failed: template: shortcodes/figure.html:125:4: executing "shortcodes/figure.html" at <partial "figure.html" $ctx>: error calling partial: "C:\mysite\node_modules\@hyas\images\layouts\partials\figure.html:204:10": execute of template failed: template: partials/figure.html:204:10: executing "partials/figure.html" at <site>: can't evaluate field landscape in type bool
```

The shortcode is using an image in the same directory as the markdown file, and the syntax is exactly the same as in the docs:

```
{{< figure
  src="vincent-van-zalinge-e5VzJJDODbQ-unsplash.jpg"
  alt="A yellow and black bird sitting on top of a body of water"
  caption="A yellow and black bird sitting on top of a body of water. Photo by Vincent van Zalinge"
>}}
```

**Reply by h-enk** [reply]
This is how I did it for Doks: https://github.com/gethyas/doks-core/blob/main/layouts/blog/single.html#L11-L23

**Reply by pensivedog** [reply]
Thank you for the reply. It seems no matter what I do, I get the below error when running `npm run build` but not when running `npm run dev`:

```
Error: error building site: render: failed to render pages: render of "page" failed: execute of template failed: template: articles/single.html:22:15: executing "main" at <partial "picture" (dict "page" . "src" $image)>: error calling partial: "C:\mysite\node_modules\@hyas\images\layouts\partials\picture.html:202:10": execute of template failed: template: partials/picture.html:202:10: executing "partials/picture.html" at <site>: can't evaluate field landscape in type bool
```

**Reply by pensivedog** [reply]
Ok, I finally figured out what was causing this error. I had previously created a config param that turned off hyas images in dev in order to have faster build times. Removing this from config and references to it in my templates got rid of this error.
